pgjdbc/src/main/java/org/postgresql/jdbc/PgDatabaseMetaData.java

```diff
@@ -1560,21 +1563,21 @@ public ResultSet getCatalogs() throws SQLException {
     return ((BaseStatement) createMetaDataStatement()).createDriverResultSet(f, v);
   }
 
+  @Override
   public ResultSet getTableTypes() throws SQLException {
     String types[] = new String[tableTypeClauses.size()];
-    Iterator<String> e = tableTypeClauses.keySet().iterator();
     int i = 0;
-    while (e.hasNext()) {
-      types[i++] = e.next();
+    for (String s : tableTypeClauses.keySet()) {
+      types[i++] = s;
```

Comment:
Would you please convert to `tableTypeClauses.keySet().toArray(new String[0])` here?